intelmq/bots/experts/sieve/expert.py

```
@@ -256,64 +279,91 @@ def process_ip_range_match(self, key, ip_range, event):
 
-    @staticmethod
-    def process_action(action, event):
+        return self._basic_math_op_map[action.operator](date, delta).isoformat()
+
+    def process_action(self, action, event) -> Procedure:
+        name = action.__class__.__name__
         if action == 'drop':
             return Procedure.DROP
         elif action == 'keep':
             return Procedure.KEEP
-        elif action.__class__.__name__ == 'PathAction':
+        elif name == 'PathAction':
             event.path = action.path
-        elif action.__class__.__name__ == 'AddAction':
+        elif name == 'AddAction':
             if action.key not in event:
                 value = action.value
                 if action.operator != '=':
-                    value = SieveExpertBot.compute_basic_math(action, event)
+                    value = self.compute_basic_math(action, event)
                 event.add(action.key, value)
-        elif action.__class__.__name__ == 'AddForceAction':
+        elif name == 'AddForceAction':
             value = action.value
             if action.operator != '=':
-                value = SieveExpertBot.compute_basic_math(action, event)
+                value = self.compute_basic_math(action, event)
             event.add(action.key, value, overwrite=True)
-        elif action.__class__.__name__ == 'UpdateAction':
+        elif name == 'UpdateAction':
             if action.key in event:
                 value = action.value
                 if action.operator != '=':
-                    value = SieveExpertBot.compute_basic_math(action, event)
+                    value = self.compute_basic_math(action, event)
                 event.change(action.key, value)
-        elif action.__class__.__name__ == 'RemoveAction':
+        elif name == 'RemoveAction':
             if action.key in event:
                 del event[action.key]
+        elif name == 'AppendAction':
+            if action.key in event and isinstance(event[action.key], list):
```

if `event[action.key]` does not exist, nothing happens, is this intentionally?

---

It is. Nonexistent keys and non-list values are intentionally silently ignored. I'm somewhat confident in the latter option, but ambivalent on whether nonexistent keys should be created or not. If you think it would be a better design to create new fields on append, then I will change this.